kyo-data/shared/src/main/scala/kyo/internal/Environment.scala

```diff
@@ -0,0 +1,37 @@
+package kyo.internal
+
+/** Environment detection utility.
+  *
+  * Provides functionality to detect whether the JVM is running in a development environment. This information is used to control the
+  * verbosity of error messages and debugging information in Kyo's implementations.
+  *
+  * The development mode can be controlled in two ways:
+  *   1. Explicitly via the system property "-Dkyo.development-mode.enable=true"
+  *   2. Automatically by detecting SBT in the classpath
+  * }}}
+  */
+private[kyo] object Environment:
+
+    /** Determines if the execution is a development environment.
+      *
+      * This method checks the following conditions in order:
+      *
+      *   1. If system property "kyo.development-mode.enable" exists:
+      *      - Returns true if the property value is "true" (case insensitive)
+      *      - Returns false if the property value is anything else
+      *   2. If the property doesn't exist:
+      *      - Returns true if SBT is detected in the classpath (contains "org.scala-sbt")
+      *      - Returns false otherwise
+      *
+      * @return
+      *   true if running in a development environment, false otherwise
+      */
+    val isDevelopment: Boolean = inferIsDevelopment()
+
+    private[internal] def inferIsDevelopment(): Boolean =
+        sys.props.get("kyo.development-mode.enable").map(_.toLowerCase) match
```

Comment:
```suggestion
        sys.props.get("kyo.development-mode.enable").flatMap(_.toBooleanOption) match
```
?
Trivial improvement, but it may be better to not dealing with strings where possible.

Reply:
I had to revert this. If the property is set to an incorrect value, I wanted to disable the development mode. It'd be possible to implement the same logic with `toBooleanOption` but it doesn't seem it'd be more clear.